### ISSUE TITLE
Improving the DataConsistencyGroup: Example, getLatestConsistentState(), different schemes

### ISSUE THREAD

**Issue by jhktimm** [ChimeraTK/DeviceAccess#52]
These 3 tasks should not all be part of this ticket. The example depends on #51. For the getLatestConsistentState() function, there must be an idea of how to do this. And adding different schemes to consistency is more for future use. But all tasks must be kept in mind.

- Example for using the DataConsistencyGroup, maybe also an example for ReadAnyGroup. This depends on "implement push type elements in dummy backend #51"

- Implement a function getLatestConsistentState: The DataConsistencyGroup should provide means to access the values of the latest consistent state. This cannot be done through the normal buffers of the accessors, since they might already contain newer values (at least with future matching schemes, see below).

- Different schemes might be added later, in particular as follows: One value is consistent with one "index value" (just a particular process variable, e.g. an event number), if the VersionNumber is greater or equal to the VersionNumber of the index value, but smaller than the one of the next index value. The implementation is not part of this ticket, but it needs to be kept in mind while designing the the structure and interface of the DataConsistencyGroup. It will e.g. require internal buffers for the values.

**Comment by jhktimm** [comment]
#25 is the first ticket from the DataConsistencyGroup
